hrm-domain/hrm-core/jobs/search/publishToSearchIndex.ts

```diff
@@ -19,11 +19,41 @@ import { getSsmParameter } from '../../config/ssmCache';
 import { IndexMessage } from '@tech-matters/hrm-search-config';
 import { CaseService, Contact } from '@tech-matters/hrm-types';
 import { AccountSID, HrmAccountId } from '@tech-matters/types';
+import { publishSns, PublishSnsParams } from '@tech-matters/sns-client';
+import { SsmParameterNotFound } from '@tech-matters/ssm-cache';
+
+type DeleteNotificationPayload = {
+  accountSid: HrmAccountId;
+  operation: 'delete';
+  id: string;
+};
+
+type UpsertCaseNotificationPayload = {
+  accountSid: HrmAccountId;
+  operation: 'update' | 'create' | 'reindex';
+  case: CaseService;
+};
+
+type UpsertContactNotificationPayload = {
+  accountSid: HrmAccountId;
+  operation: 'update' | 'create' | 'reindex';
+  contact: Contact;
+};
```

Comment:
Since this is getting a bit complicated to follow, and scattered across repos, should we consider having a README explaining what each message is and what it's targets are? We could then link the README to the relevant parts of the code. I am thinking about myself in 2 weeks to be honest :laughing: 
PS: this suggestion does not needs to be address in this PR, but having a ticket to track that might be desirable?

Reply:
We should think of SNS broadcasts like an API. where the broadcaster (HRM) is the service and the subscribers are clients.

Just like an API I don't think we want to document both sides of the API all together, but we do want to ensure the API 'contract' (i.e. the message attributes) are documented in the repo that owns the broadcast - i.e. HRM